test/test_spectral_ops.py

```diff
@@ -322,11 +294,9 @@ def test_fn(x):
 
 
     @slowAwareTest
-    @skipCPUIfNoMkl
-    @skipCUDAIfRocm
     @onlyOnCPUAndCUDA
-    @dtypes(torch.double, torch.complex128)  # gradcheck requires double
-    def test_fft_backward(self, device, dtype):
+    @ops(spectral_funcs, allowed_dtypes=[torch.double, torch.complex128])  # gradcheck requires double
+    def test_backward_1d(self, device, dtype, op):
```

Comment:
This is interesting. There are existing grad and gradgrad checks using the sample inputs in test_ops.py. See, for example:

https://github.com/pytorch/pytorch/blob/bcc85a363ecbcb381623d3c26faa94be13ad58bc/test/test_ops.py#L98

Can those tests be used exclusively to test the backward methods by generating the right sample inputs?

Reply:
Sure, it would just mean generating a lot more sample inputs. The sample lists for other operators seem to be very minimal so I wasn't sure what the intention is with that.

Reply:
Good point - let's add a comment describing the purpose of sample inputs better. Basically it should be all inputs we think are necessary to check that a function's grad and gradgrad work properly, so these tests can be folded in.

Reply:
@mruberry I've added a comment to the `sample_inputs` docstring, added more sample inputs, and removed the backwards tests from `test_spectral_ops.py`.